src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -498,7 +565,7 @@ private static Exception InvalidTypeException(object value, string argument)
         }
 
         [Serializable]
-        private class Enumerator : IEnumerator<T>, IEnumerator
+        private struct Enumerator : IEnumerator<T>, IEnumerator
```

Comment:
This offers a performance boost if the type is public and the `GetEnumerator()` typed as returning it, though as well as being a change to the public API it's an optimisation that has caveats. As is this doesn't gain anything.